[iOS] Canvas and CanvasText should be implemented as system colors

pxlcoder opened the pull request:
#### 2b6e8dd0c7574131afcfcf2e9767a289135c66ac
<pre>
[iOS] Canvas and CanvasText should be implemented as system colors
<a href="https://bugs.webkit.org/show_bug.cgi?id=232953">https://bugs.webkit.org/show_bug.cgi?id=232953</a>
rdar://85510671

Reviewed by Tim Nguyen.

Canvas and CanvasText are already implemented as system colors on macOS.
These colors should also be implemented as system colors on iOS, matching
the spec.

Spec: <a href="https://www.w3.org/TR/css-color-4/">https://www.w3.org/TR/css-color-4/</a>

* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-attribute-changes-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-empty-content-value-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-first-valid-applies-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-insert-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-no-content-value-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-remove-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-remove-head-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-body-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-head-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-shadow-tree-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/default-color-expected.txt: Removed.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::cssValueSystemColorInformationList):

&apos;Canvas&apos; is a system color which represents &quot;the background of
application content or documents&quot;. This corresponds to
`+[UIColor systemBackgroundColor]` on iOS.

&apos;CanvasText&apos; is a system color which represents &quot;text in
application content or documents&quot;. This corresponds to
`+[UIColor labelColor]` on iOS.

Canonical link: <a href="https://commits.webkit.org/252718@main">https://commits.webkit.org/252718@main</a>
</pre>

pxlcoder commented on LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-attribute-changes-expected.txt:
```diff
@@ -1,10 +1,10 @@
-FAIL Set content attribute of meta color-scheme to 'light'. assert_not_equals: CanvasText system color should be different with light and dark color schemes got disallowed value "rgb(0, 0, 0)"
-FAIL Set content attribute of meta color-scheme to 'dark'. assert_not_equals: CanvasText system color should be different with light and dark color schemes got disallowed value "rgb(0, 0, 0)"
-FAIL Removed the content attribute of meta color-scheme. assert_not_equals: CanvasText system color should be different with light and dark color schemes got disallowed value "rgb(0, 0, 0)"
+FAIL Meta color-scheme initially 'dark'. assert_equals: Root element's color-scheme should be 'normal' expected "normal" but got "auto"
```
The rebaselines in this folder are net progressions, the tests fail later than they used to.

The remaining failures are due to https://bugs.webkit.org/show_bug.cgi?id=224810.

nt1m reviewed (approved):
If the iOS baselines now match the non-iOS ones, can we delete them?

pxlcoder committed:
[iOS] Canvas and CanvasText should be implemented as system colors
https://bugs.webkit.org/show_bug.cgi?id=232953
rdar://85510671

Reviewed by Tim Nguyen.

Canvas and CanvasText are already implemented as system colors on macOS.
These colors should also be implemented as system colors on iOS, matching
the spec.

Spec: https://www.w3.org/TR/css-color-4/

* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-attribute-changes-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-empty-content-value-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-first-valid-applies-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-insert-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-no-content-value-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-remove-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-remove-head-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-body-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-head-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/document-metadata/the-meta-element/color-scheme/meta-color-scheme-single-value-in-shadow-tree-expected.txt: Removed.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/default-color-expected.txt: Removed.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::cssValueSystemColorInformationList):

'Canvas' is a system color which represents "the background of
application content or documents". This corresponds to
`+[UIColor systemBackgroundColor]` on iOS.

'CanvasText' is a system color which represents "text in
application content or documents". This corresponds to
`+[UIColor labelColor]` on iOS.

Canonical link: https://commits.webkit.org/252718@main
